src/components/content/order/CreateService.tsx

```diff
@@ -102,49 +105,83 @@ function CreateService(): JSX.Element {
+        if (categoryName === '' || serviceName === '' || latestVersion === '') {
             return;
         }
         setCategoryName(categoryName);
         setServiceName(serviceName);
-        void serviceVendorApi.listRegisteredServices(categoryName, '', serviceName, '').then((rsp) => {
-            if (rsp.length > 0) {
-                const currentVersions: Map<string, RegisteredServiceVo[]> = new Map<string, RegisteredServiceVo[]>();
-                for (const registerServiceEntity of rsp) {
-                    if (registerServiceEntity.version) {
-                        if (!currentVersions.has(registerServiceEntity.version)) {
-                            currentVersions.set(
-                                registerServiceEntity.version,
-                                rsp.filter((data) => data.version === registerServiceEntity.version)
-                            );
+        void serviceVendorApi
+            .listRegisteredServices(categoryName, '', serviceName, '')
+            .then((rsp) => {
+                if (rsp.length > 0) {
+                    const currentVersions: Map<string, RegisteredServiceVo[]> = new Map<
+                        string,
+                        RegisteredServiceVo[]
+                    >();
+                    for (const registerServiceEntity of rsp) {
+                        if (registerServiceEntity.version) {
+                            if (!currentVersions.has(registerServiceEntity.version)) {
+                                currentVersions.set(
+                                    registerServiceEntity.version,
+                                    rsp.filter((data) => data.version === registerServiceEntity.version)
+                                );
+                            }
                         }
                     }
+                    versionMapper.current = currentVersions;
+                    const currentVersionList = getVersionList();
+                    const currentCspList = getCspList(latestVersion);
+                    const currentFlavorList = getFlavorList(latestVersion);
+                    setVersionList(currentVersionList);
+                    setSelectVersion(latestVersion);
+                    setCspList(currentCspList);
+                    setFlavorList(currentFlavorList);
+                    let currentAreaList: Tab[] = getAreaList(latestVersion, currentCspList[0]);
+                    let currentRegionList: { value: string; label: string }[] = getRegionList(
+                        latestVersion,
+                        currentCspList[0],
+                        currentAreaList[0] ? currentAreaList[0].key : ''
+                    );
+                    let currentBilling = getBilling(latestVersion, currentCspList[0]);
+                    let cspValue: CloudServiceProviderNameEnum = currentCspList[0];
+                    let areaValue: string = currentAreaList[0] ? currentAreaList[0].key : '';
```

Comment:
```suggestion
                   let areaValue: string = currentAreaList[0]?.key ?? '';
```
we can make it simpler like this.